pkg/cmd/server/api/validation/oauth.go

```diff
@@ -162,6 +166,24 @@ func ValidateLDAPIdentityProvider(provider *api.LDAPPasswordIdentityProvider) Va
 	return validationResults
 }
 
+// RemoteConnection fields validated separately -- this is for keystone-specific validation
+func ValidateKeystoneIdentityProvider(provider *api.KeystonePasswordIdentityProvider, identityProvider api.IdentityProvider) ValidationResults {
+	validationResults := ValidationResults{}
+	validationResults.AddErrors(ValidateRemoteConnectionInfo(provider.RemoteConnectionInfo)...)
+
+	providerURL, err := url.Parse(provider.RemoteConnectionInfo.URL)
+	if err == nil {
+		if providerURL.Scheme != "https" {
+			validationResults.AddWarnings(fielderrors.NewFieldInvalid("url", provider.RemoteConnectionInfo.URL, "Auth URL should be secure and start with https"))
+		}
+	}
```

Comment:
I forgot, `ValidateSecureURL` already exists... I think this whole bit can simplify to 
```
validationResults.AddWarnings(ValidateSecureURL("url", provider.RemoteConnectionInfo.URL)
```